torch/lib/c10d/ProcessGroup.hpp

```diff
@@ -162,6 +162,22 @@ class ProcessGroup {
       std::vector<std::vector<at::Tensor>>& inputTensors,
       const ReduceScatterOptions& opts = ReduceScatterOptions()) = 0;
 
+  virtual std::shared_ptr<ProcessGroup::Work> alltoall_base(
+      std::vector<at::Tensor>& outputTensors,
+      std::vector<at::Tensor>& inputTensors,
+      std::vector<int>& outputSplitSizes,
+      std::vector<int>& inputSplitSizes,
+      const AllToAllOptions& opts = AllToAllOptions()) {
+      throw std::runtime_error("ProcessGroup does not support alltoall");
+  }
+
+  virtual std::shared_ptr<ProcessGroup::Work> alltoall(
+      std::vector<std::vector<at::Tensor>>& outputTensors,
+      std::vector<std::vector<at::Tensor>>& inputTensors,
+      const AllToAllOptions& opts = AllToAllOptions()) {
+      throw std::runtime_error("ProcessGroup does not support alltoall");
+  }
```

Comment:
And keep this as it is?

Reply:
@agolynski what do you think about the API?

Reply:
i'd keep a API to a necessary minimum for now. We can add alltoall_multigpu if necessary later (note that multigpu API is quite complex, see below)? 

Some observations:
1) vector<vector<>> for inputs might be confusing as our APIs suggest that we are doing this op on multiple tensors and we don't plan to have alltoall_coalesced as yet (right?)
2) outputs: I believe we use outward vector<> to index workers in the process group and inward one to index tensors by group index, gpu index, e.g.
https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L1081
This API is quite complex and took sometime to get my head around it, and it's not necessary for users to understand that type of indexing for alltoall to work, right?
3) if we decide to go with vector<vector<>> API for outputs, I believe you used inward vector<> to index index workers in the process group and outward vector<> was presumed to have one element. Would we need to change this for consistency?

Reply:
Sounds good, I will change this to use just `vector<Tensor>` for both input and output.